### ISSUE TITLE
sample_frac wrong on grouped data without indices attribute (e.g. after a summarise)

### ISSUE THREAD

**Issue by tmelconian** [tidyverse/dplyr#3710]
This works as I expect:
```r
> expand.grid(a=1:5,b=1:5) %>% group_by(a,b) %>% summarise %>% group_by(a) %>% sample_frac(0.2)
# A tibble: 5 x 2
# Groups:   a [5]
      a     b
  <int> <int>
1     1     1
2     2     4
3     3     1
4     4     3
5     5     4
```

I should be able to remove the group_by(a) in the call above, because coming out of the summarise(), it's already grouped by a.  But something goes very wrong when I do so:

```r
> expand.grid(a=1:5,b=1:5) %>% group_by(a,b) %>% summarise %>% sample_frac(0.2)
# A tibble: 0 x 2
# Groups:   a [0]
# ... with 2 variables: a <int>, b <int>
```

The only difference between the two cases is the absence of the "indices" attribute in the former case.  We can verify that stripping indices reproduces the broken behavior:

```r
> expand.grid(a=1:5,b=1:5) %>% group_by(a,b) %>% summarise %>% group_by(a) %>% structure(indices=NULL) %>% sample_frac(0.2)
# A tibble: 0 x 2
# Groups:   a [0]
# ... with 2 variables: a <int>, b <int>
```

**Comment by jmarshallnz** [comment]
Just a quick note that this is working fine with 0.7.99.9000 :)

**Comment by tmelconian** [comment]
Ah, yes, I should have mentioned the version; I'm using 0.7.6, which is the latest released version available on CRAN.  I am not in a position to run development releases, but if you want to close the bug because it's fixed in development feel free. :)

**Comment by jmarshallnz** [comment]
Yup, the way grouping info is stored was changed for 0.8 so once that hits CRAN this will be resolved :)

**Comment by krlmlr** [comment]
Thanks. Confirmed in v0.7.6, fixed in `master`. What about v0.7.5?

**Comment by tmelconian** [comment]
I first observed it was broken in 0.7.4, so almost certainly it was also broken in 0.7.5.

**Comment by krlmlr** [comment]
Thanks, in this case I don't think we should be doing anything in v0.7.6.

**Comment by romainfrancois** [comment]
This has been fixed as part of #3526 or one of its neighbor issues. `summarise` and the other verbs no longer make lazy grouped tibbles.
